pkg/modifycontroller/controller.go

```diff
@@ -0,0 +1,334 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modifycontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/kubernetes-csi/external-resizer/pkg/features"
+	"github.com/kubernetes-csi/external-resizer/pkg/util"
+
+	"github.com/kubernetes-csi/external-resizer/pkg/modifier"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+// ModifyController watches PVCs and checks if they are requesting an modify operation.
+// If requested, it will modify the volume according to parameters in VolumeAttributesClass
+type ModifyController interface {
+	// Run starts the controller.
+	Run(workers int, ctx context.Context)
+}
+
+type modifyController struct {
+	name            string
+	modifier        modifier.Modifier
+	kubeClient      kubernetes.Interface
+	claimQueue      workqueue.RateLimitingInterface
+	vacQueue        workqueue.RateLimitingInterface
+	eventRecorder   record.EventRecorder
```

Comment:
The `vacQueue` is not being used and the entire mechanism of watching vac here can be removed, if we are not going to do anything with it.

Reply:
To be clear we still need to wait for vacs to be synced and we need a lister, but we don't need `vacAdded` etc functions, if we are not going to do anything with them.

Reply:
Removed